### ISSUE TITLE
[TFS 2018] Build definition imported partially

### ISSUE THREAD

**Issue by eduardohass** [DarqueWarrior/vsteam#78]
I tried to import new build definition to my TFS 2018(update 2) On Premise using the VsTeam module and i have an issue past execution (success executiuoin

Powershell command used to import build defintion to TFS

```powershell
Add-VSTeamBuildDefinition -ProjectName "project-test" -InFile "bd-project-test-template.json"
````

This (bellow) is a fragment of json used to import build definition (phases) to TFS:

```json
....
"process": {
    "phases": [
        {
            "steps": [
                {
                    "environment": {},
                    "enabled": true,
                    "continueOnError": false,
                    "alwaysRun": false,
                    "displayName": "Restore",
                    "timeoutInMinutes": 0,
                    "task": {
                        "id": "ffd7bc7f-a32c-4b4b-a52d-ebfdb60046bc",
                        "versionSpec": "1.*",
                        "definitionType": "task"
                    },
                    "inputs": {
                        "command": "restore",
                        "publishWebProjects": "true",
                        "projects": "$(Parameters.RestoreBuildProjects)",
                        "arguments": "--configfile nuget.config --force",
                        "zipAfterPublish": "true"
                    }
                },
                {
                    "environment": {},
                    "enabled": true,
                    "continueOnError": false,
                    "alwaysRun": false,
                    "displayName": "Build",
                    "timeoutInMinutes": 0,
                    "task": {
                        "id": "ffd7bc7f-a32c-4b4b-a52d-ebfdb60046bc",
                        "versionSpec": "1.*",
                        "definitionType": "task"
                    },
                    "inputs": {
                        "command": "build",
                        "publishWebProjects": "true",
                        "projects": "$(Parameters.RestoreBuildProjects)",
                        "arguments": "--configuration $(BuildConfiguration) --no-restore",
                        "zipAfterPublish": "true"
                    }
                },
                {
                    "environment": {},
                    "enabled": true,
                    "continueOnError": false,
                    "alwaysRun": false,
                    "displayName": "Test",
                    "timeoutInMinutes": 0,
                    "task": {
                        "id": "ffd7bc7f-a32c-4b4b-a52d-ebfdb60046bc",
                        "versionSpec": "1.*",
                        "definitionType": "task"
                    },
                    "inputs": {
                        "command": "test",
                        "publishWebProjects": "true",
                        "projects": "$(Parameters.TestProjects)",
                        "arguments": "--configuration $(BuildConfiguration)",
                        "zipAfterPublish": "true"
                    }
                }
            ],
            "name": "Phase 1",
            "refName": "Phase_1",
            "target": {
                "executionOptions": null,
                "allowScriptsAuthAccessOption": false,
                "type": 1
            },
            "jobAuthorizationScope": 0
        }
    ],
    "type": 1
},
...
```

Expected behavior
---------------
![01_bd_project_test_excpected](https://user-images.githubusercontent.com/4368558/44271816-8a1b7880-a211-11e8-997e-67295dde6fa7.png)

Actual behavior
---------------
![01_bd_project_test](https://user-images.githubusercontent.com/4368558/44271559-d5815700-a210-11e8-94ca-69ac318d897f.png)

Obs: I tried to use New-Guid, empty GUID, removed the GUID line, etc.

**Comment by DarqueWarrior** [comment]
How large is the json file on disk? I wonder if there is a payload size I need to adjust in the post call or something.

**Comment by eduardohass** [comment]
My files have 10kb.
I made a test with a release definition with 50kb and was successful.

**Comment by japj** [comment]
The multiphase support in the rest api was done around 3.2 I believe.
So You probably need to set relevant VSTeamVersions item to higher than 3.2 (default is currently 3.0).
So even if your json file is of 4.0 structure, the default behavior will be to push to the 3.0 api, which does not have multiphase support and as a result “breaks” the build definition.

**Comment by DarqueWarrior** [comment]
You can change the API version used.  Run 

```PowerShell
help about_vsteam
```

To learn how to change the API version used.

**Comment by DarqueWarrior** [comment]
Thanks, @japj I was missing that point about multi-phase builds. I will work on a PR to update the version used in Azure DevOps.

**Comment by MichelZ** [comment]
Is this still an issue?

**Comment by SebastianSchuetze** [comment]
This should be fixed now with PR #321 which allows also to target updates of servers @eduardohass.
